meilisearch/tests/vector/openai.rs

```diff
@@ -0,0 +1,1873 @@
+        settings["model"] = serde_json::json!(self.model());
+        if let ModelDimensions::Large512 = self {
+            settings["dimensions"] = serde_json::json!(512);
+        }
+    }
+
+    fn model(&self) -> &'static str {
+        match self {
+            ModelDimensions::Large | ModelDimensions::Large512 => "text-embedding-3-large",
+            ModelDimensions::Small => "text-embedding-3-small",
+            ModelDimensions::Ada => "text-embedding-ada-002",
+        }
+    }
+
+    fn from_request(request: &serde_json::Value) -> Self {
+        let has_dimensions_512 = if let Some(dimensions) = request.get("dimensions") {
+            if dimensions != 512 {
+                panic!("unsupported dimensions values")
+            }
+            true
+        } else {
+            false
+        };
+        let serde_json::Value::String(model) = &request["model"] else {
+            panic!("unsupported non string model")
+        };
+        match (model.as_str(), has_dimensions_512) {
+            ("text-embedding-3-large", true) => Self::Large512,
+            (_, true) => panic!("unsupported dimensions with non-large model"),
+            ("text-embedding-3-large", false) => Self::Large,
+            ("text-embedding-3-small", false) => Self::Small,
+            ("text-embedding-ada-002", false) => Self::Ada,
+            (_, false) => panic!("unsupported model"),
+        }
+    }
+}
+
+fn openai_responses() -> &'static OpenAiResponses {
+    static OPENAI_RESPONSES: OnceLock<OpenAiResponses> = OnceLock::new();
+    OPENAI_RESPONSES.get_or_init(|| {
+        // json file that was compressed with gzip
+        // decompress with `gzip --keep -d openai_responses.json.gz`
+        // recompress with `gzip --keep -c openai_responses.json > openai_responses.json.gz`
+        let compressed_responses = include_bytes!("openai_responses.json.gz");
+        let mut responses = Vec::new();
+        let mut decoder = flate2::write::GzDecoder::new(&mut responses);
+
+        decoder.write_all(compressed_responses).unwrap();
+        drop(decoder);
+        serde_json::from_slice(&responses).unwrap()
+    })
+}
+
+fn openai_tokenized_responses() -> &'static OpenAiTokenizedResponses {
+    static OPENAI_TOKENIZED_RESPONSES: OnceLock<OpenAiTokenizedResponses> = OnceLock::new();
+    OPENAI_TOKENIZED_RESPONSES.get_or_init(|| {
+        // json file that was compressed with gzip
+        // decompress with `gzip --keep -d openai_tokenized_responses.json.gz`
+        // recompress with `gzip --keep -c openai_tokenized_responses.json > openai_tokenized_responses.json.gz`
+        let compressed_responses = include_bytes!("openai_tokenized_responses.json.gz");
+        let mut responses = Vec::new();
+        let mut decoder = flate2::write::GzDecoder::new(&mut responses);
+
+        decoder.write_all(compressed_responses).unwrap();
+        drop(decoder);
+        serde_json::from_slice(&responses).unwrap()
```

Comment:
We may consider putting an on-disk cache on this part if it starts taking a lot of time